### DISCUSSION TITLE
[Vue warn]: The `compilerOptions` config option...

### DISCUSSION THREAD

**Discussion by CrawfordW** [quasarframework/quasar#15712] [General - Components / Directives / etc]
Suddenly started getting lots of these in the devtools console...
[Vue warn]: The `compilerOptions` config option is only respected when using a build of Vue.js that includes the runtime compiler (aka "full build"). Since you are using the runtime-only build, `compilerOptions` must be passed to `@vue/compiler-dom` in the build setup instead.

I have not changed my project configuration - just the way quasar scaffolded it using webPack. Upgraded to latest quasar version  but still get these errors/warnings.

**Comment by yusufkandemir** [comment]
We'd need more context, but you probably do stuff like `app.compilerOptions.whatever = ''` in a boot file or similar. You should use `quasar.config.js > vueLoaderOptions > compilerOptions` instead.

You can search for `compilerOptions` in your project to find if it is misused anywhere. If you don't have it in your code, then you can check the Vue plugins you use. Remove `app.use` statements one by one to find the faulty one, then seek the repo/documentation/issues/etc. for the plugin and try to find a solution.

**Reply by CrawfordW** [reply]
The only compilerOptions I found is in the jsconfig.json file which I presume came from scaffolding the project with quasar cli.
the quasar.config.js file has no vueLoaderOptions.

**Reply by CrawfordW** [reply]
My project has many SFCs written using the options API. I'm trying to learn the composition API, and my first SFC attempt generates these error messages as soon as I select it in the Vue Devtools component list. I don't see the errors anywhere else.
Following is the list of imports for this SFC (most are used in many other SFCs except the ones needed only for composition API):
<script setup>
import SelectTree from '../widgets/SelectTree.vue'
import WorkPanelHeader from '../widgets/WorkPanelHeader'
import ToolbarButton from '../widgets/ToolbarButton'
import SelectAutofill from '../widgets/SelectAutofill'
import NewScrollArea from '../widgets/NewScrollArea'
import TableValue from '../widgets/TableValue.vue'
import { useWorkTabsStore } from 'stores/WorkTabsStore'
import { useSuppliesStore } from 'stores/SuppliesStore'
import { useSettingsStore } from 'stores/SettingsStore'
import { getItemIdFromName } from 'stores/commonGetters'
import cloneDeep from 'lodash.clonedeep'
import { convertUnits, formatCurrency, formatSignificant } from 'src/common/commonFunctions'

//  These are needed with composition API:
import { ref, reactive, computed, useAttrs, onMounted, onBeforeMount } from 'vue' // , watch
import { useQuasar } from 'quasar'

**Reply by CrawfordW** [reply]
Sure wish I could get some help with this. Not using any Vue plugins. Only Quasar plugins are:
'AppFullscreen',  'Dialog',  'LocalStorage', and  'Loading'
Only 'compilerOptions' is in jsconfig.json file:
"compilerOptions": {
    "baseUrl": ".",
    "paths": {
      "src/*": [
        "src/*"
      ],
      "app/*": [
        "*"
      ],
      "components/*": [
        "src/components/*"
      ],
      "layouts/*": [
        "src/layouts/*"
      ],
      "pages/*": [
        "src/pages/*"
      ],
      "assets/*": [
        "src/assets/*"
      ],
      "boot/*": [
        "src/boot/*"
      ],
      "stores/*": [
        "src/stores/*"
      ],
      "vue$": [
        "node_modules/vue/dist/vue.runtime.esm-bundler.js"
      ]
    }
  },
